x/protorev/keeper/epoch_hook.go

```diff
@@ -0,0 +1,118 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochstypes "github.com/osmosis-labs/osmosis/v13/x/epochs/types"
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+type EpochHooks struct {
+	k Keeper
+}
+
+// Struct used to track the pool with the highest liquidity
+type LiquidityPoolStruct struct {
+	Liquidity sdk.Int
+	PoolId    uint64
+}
+
+var (
+	_ epochstypes.EpochHooks = EpochHooks{}
+)
+
+func (k Keeper) EpochHooks() epochstypes.EpochHooks {
+	return EpochHooks{k}
+}
+
+// BeforeEpochStart is the epoch start hook.
+func (h EpochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook. The module will update all of the pools in the store that are
+// used for trading.
+func (h EpochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	switch epochIdentifier {
+	case "week":
+		// Reset the pools in the store
+		h.k.DeleteAllAtomPools(ctx)
```

Comment:
Why have separate functions for delete and update? Would we ever call delete without updating or vice versa?

Reply:
I was thinking that we reset all of the pools and have a clean slate when setting the pools in the `UpdatePools` function. I wasn't fully sure if pools ever get removed for whatever reason but it seems like thats an edge case I don't need to account for (so I will remove this deletion code and have it only update).

Reply:
Per the comment above this one, the epoch hook will also be populated with functionality that distributes a portion of the developer fees to a developer account (which presumably will be skip's address once deployed). This requires us to track various info such as the number of days since genesis so that we can properly split the arbitrage revenue generated at any given moment in time.